### PR TITLE
Add Deathmatch event

### PR DESCRIPTION
**Pull request by matiascalegaris**

### CODE REVIEW

File: Codigo/CustomScenarios.bas [hunk 1]
```diff
@@ -123,9 +128,25 @@ On Error GoTo PrepareNewEvent_Err:
     Select Case EventType
         Case e_EventType.NpcHunt
             Set GenericGlobalLobby.scenario = New ScenarioHunt
+        Case e_EventType.DeathMatch
+            Set GenericGlobalLobby.scenario = New ScenarioDeathMatch
```
Comment (morgolock) [inline Codigo/CustomScenarios.bas:132]: missing `case else`

File: Codigo/ScenarioDeathMatch.cls [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+108    CustomScenarios.RemoveUpdateScenario (UpdateIndex)
+110    MapInfo(MapNumber).DropItems = MapDropItems
+112    MapInfo(MapNumber).Seguro = PrevMapState
+114    MapInfo(mapNumber).SafeFightMap = SafeFigthState
+    Exit Sub
+CloseScenario_Err:
+    Call TraceError(Err.Number, Err.Description, "ScenarioHunt.CloseScenario ", Erl)
+End Sub
+
+Private Function IsForbidden(itemId As Integer) As Boolean
```
Comment (morgolock) [inline Codigo/ScenarioDeathMatch.cls:109]: `byval itemID`

File: Codigo/ScenarioKillNpc.cls [hunk 3]
```diff
@@ -111,7 +120,7 @@ Private Function IBaseScenario_ValidateUser(ByVal userIndex As Integer) As Integ
                 Exit Function
             End If
         Next i
-        IBaseScenario_ValidateUser = 403 'TODO create valid message here
+        IBaseScenario_ValidateUser = 403
```
Comment (morgolock) [inline Codigo/ScenarioKillNpc.cls:123]: What's `403`? can we please move it to a named `const variable` ?

File: Codigo/ScenarioKillNpc.cls [hunk 4]
```diff
@@ -121,17 +130,15 @@ Private Sub IBaseScenario_WillSummonPlayer(ByVal UserIndex As Integer)
...
+    Call board.AddPlayer(UserIndex)
 End Sub
 
 Private Sub IBaseScenario_UpdateLobbyState(ByVal oldState As e_LobbyState, ByRef newState As e_LobbyState)
     If oldState = newState Then
         Exit Sub
     End If
     If newState = InProgress Then
-        Call StartGame
+        Call SetTimer(StartTimer, 1000)
```
Comment (morgolock) [inline Codigo/ScenarioKillNpc.cls:141]: replace `1000` by a named `CONST` variable

**Review by morgolock — CHANGES_REQUESTED** [review]

File: Codigo/CustomScenarios.bas (hunk 1 above)
Comment (matiascalegaris) [reply Codigo/CustomScenarios.bas:132]: there is nothing to do in an else case, this is intended, its  fine to do nothing if there is no event attached to a lobby

**Review by morgolock — APPROVED** [review]

**Pull request merged**